### PR TITLE
for `plotImage`, add argument to control contrast (`cl`), including doc examples, unit tests etc.

### PR DESCRIPTION
**Pull request by HelenaLC**

### CODE REVIEW

File: R/plotImage.R [hunk 1]
```diff
@@ -49,9 +55,12 @@ plotSpatialData <- \() ggplot() + scale_y_reverse() + .theme
         c <- col2rgb(c)/255
         b <- array(0, dim=c(3, dim(a)[-1]))
         for (i in seq_len(dim(a)[1])) {
-            b[1,,] <- b[1,,,drop=FALSE] + a[i,,,drop=FALSE]*c[1,i]
-            b[2,,] <- b[2,,,drop=FALSE] + a[i,,,drop=FALSE]*c[2,i]
-            b[3,,] <- b[3,,,drop=FALSE] + a[i,,,drop=FALSE]*c[3,i]
+            b[1,,] <- b[1,,,drop=FALSE] + a[i,,,drop=FALSE]*c[1,i]*(1/lim[[i]][2])*sat[i]
```
Comment (HelenaLC) [inline R/plotImage.R]: I am actually not so sure about the `sat` part... seems redundant as is, so maybe it should be applied differently? (1/lim) makes sense to me and is also what I found in imaging literature for how contrast limits work.

**Comment by HelenaLC** [comment]
@Artur-man general Q: `.manage_channels` is currently only entered when the image isn't an RGB. But with contrast limits in place, don't we also wanna go there? E.g., for the blobs, I currently can't alter how the image is rendered at all.

**Comment by Artur-man** [comment]
Hmmmm, I have never thought about this. I used to only change the saturation and brightness of IF channels before. Perhaps we can deal with this when channels are passed to .ch2rgb, hence pass rgb images there too ? then use .is_rgb function only in .ch2rgb ? I dont know

**Comment by HelenaLC** [comment]
changing this line in `.df_i()` seems to do the trick:
```
#if (!.is_rgb(x))
if (!is.null(c) || !is.null(cl))
    a <- .chs2rgb(a, ch_i, c, cl)
    
x <- file.path("extdata", "blobs.zarr")
x <- system.file(x, package="SpatialData")
x <- readSpatialData(x, tables=FALSE)
plotSpatialData() + plotImage(x, c=c("cyan", "magenta", "yellow"))
```
image

**Comment by Artur-man** [comment]
```
if (!is.null(c) || !is.null(cl))
    a <- .chs2rgb(a, ch_i, c, cl)
```

Hmm ... so lets say that colors are null but contrast limits are not, then we go to chsrgb .... if there are colors and contrast limits are null, then we go to chsrgb again, if both are null we go chrsrgb ... 

so the only case we dont go to chs2rgb is when both colors and cl are null .... but then the statement before we already pick the first channel when everything is null

```
if (!.is_rgb(x))
        a <- a[ch_i, , , drop=FALSE]
```

ok you are right this is ok!

**Comment by HelenaLC** [comment]
I realize we need both - `!.is_rgb(x) || !is.null(c) || !is.null(cl)` - else one-channel (e.g., MERFISH) fails. I also included unit tests now. Updating soon.

**Comment by HelenaLC** [comment]
Alright, so actually still need the `!.is_rgb`, else single-channel images will fail. I also included unit tests now. Let me know if we're happy with this (for now) & I'll merge.

**Pull request merged**